shell/platform/fuchsia/flutter/surface_producer.h

```diff
@@ -0,0 +1,50 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_FUCHSIA_FLUTTER_SURFACE_PRODUCER_H_
+#define FLUTTER_SHELL_PLATFORM_FUCHSIA_FLUTTER_SURFACE_PRODUCER_H_
+
+#include <functional>
+#include <memory>
+#include <vector>
+
+#include "third_party/skia/include/core/SkSize.h"
+#include "third_party/skia/include/core/SkSurface.h"
+
+namespace flutter_runner {
+
+class SurfaceProducerSurface {
```

Comment:
I know this PR just moves existing code, but adding a brief comment on what each class does would be very nice.